### DISCUSSION TITLE
migration example?

### DISCUSSION THREAD

**Discussion by degenaro** [pydantic/pydantic#9333] [Question]
V1:
```
class UUIDDatatype(OscalBaseModel):
    __root__: constr(regex=r'^[0-9A-Fa-f]{8}-[0-9A-Fa-f]{4}-[45][0-9A-Fa-f]{3}-[89ABab][0-9A-Fa-f]{3}-[0-9A-Fa-f]{12}$') = Field(..., description="A type 4 ('random' or 'pseudorandom') or type 5 UUID per RFC 4122.")
```
V2:
`? `

Or is there a pointer to something?
Thx.

**Comment by tomschr** [comment]
Hi @degenaro,

I'm not an expert, but I stumbled upon the `constr` function as well. According to the docstring, it's deprecated and shouldn't be used. The docstring says, you should use `Annotated` and `StringConstraints` as explained in the [API documentation](https://docs.pydantic.dev/latest/api/types/#pydantic.types.constr).

You could start with this:

```python
class UUIDDatatype(OscalBaseModel):
   __root__: Annotated[str, StringConstraints(pattern=r'...')]
```

Not sure about the `Field`. I hope it gives you at least a rough idea to proceed. Good luck!
